### PR TITLE
User Email Updates

### PR DESCRIPTION
**Pull request by jSylvestre**

### CODE REVIEW

File: Keas.Core/Services/IUpdateFromIamService.cs [hunk 1]
```diff
@@ -140,6 +151,16 @@ public async Task<int> UpdateAllUsersFromIam()
...
                         }
+                        if(!string.IsNullOrWhiteSpace(ietData.CampusEmail) && user.Email != ietData.CampusEmail)
+                        {
+                            user.Email = ietData.CampusEmail;
+                            Log.Information($"Updating {user.Iam} email from Iam.");
+                        }
+                        if (string.IsNullOrWhiteSpace(ietData.CampusEmail))
+                        {
+                            //Investigate these
+                            Log.Information($"Missing CampusEmail IAM: {user.Iam}");
```
Comment (sprucely) [inline Keas.Core/Services/IUpdateFromIamService.cs:162]: Should this be a warning?

**Review by sprucely — APPROVED** [review]
Didn't test, but looks fine.

File: Keas.Mvc/Services/IdentityService.cs [hunk 2]
```diff
@@ -228,8 +219,38 @@ public async Task<User> GetByKerberos(string kerb)
...
+                var ucdContact = ucdContactResult.ResponseData.Results.Where(a => a.Email != null).FirstOrDefault();
+                if (ucdContact != null)
+                {
+                    //Or try to get the contact email
+                    return ucdContact.Email;
+                }
+            }
+
+            //Or just use the kerb with email. Might be ok.
+            return $"{ucdKerbPerson.UserId}@ucdavis.edu";
```
Comment (laholstege) [inline Keas.Mvc/Services/IdentityService.cs:250]: is this something we normally do? this isn't a safe bet. my kerb@ucdavis.edu would be wrong

File: Keas.Core/Services/IUpdateFromIamService.cs (hunk 1 above)
Comment (jSylvestre) [reply Keas.Core/Services/IUpdateFromIamService.cs:162]: Yeah... Probably

File: Keas.Mvc/Services/IdentityService.cs [hunk 3]
```diff
@@ -198,26 +198,17 @@ public async Task<User> GetByKerberos(string kerb)
...
             }
 
-            var ucdContact = ucdContactResult.ResponseData.Results.First();
-            var rtValue = CreateUser(ucdContact.Email, ucdKerbPerson, ucdKerbPerson.IamId);
-
-            if (string.IsNullOrWhiteSpace(rtValue.Email))
-            {
-                if (!string.IsNullOrWhiteSpace(ucdKerbPerson.UserId))
-                {
-                    rtValue.Email = $"{ucdKerbPerson.UserId}@ucdavis.edu";
```
Comment (jSylvestre) [inline Keas.Mvc/Services/IdentityService.cs:218]: @laholstege It was doing here.

Comment (laholstege) [reply Keas.Mvc/Services/IdentityService.cs:218]: alright, well. i think it'd be better to have no email than one we're guessing on, but i don't feel strongly about it and if we were doing it before i guess it's fine

**Review by laholstege — APPROVED** [review]

**Pull request merged**